src/main/java/com/simpleftp/ui/containers/LocalFilePanelContainer.java

```diff
@@ -0,0 +1,174 @@
+            }
+        } catch (IOException ex) {
+            UI.doException(ex, UI.ExceptionType.EXCEPTION, FTPSystem.isDebugEnabled());
+        }
+    }
+
+    /**
+     * Handler for creating a local directory
+     * @param path the path for the directory
+     * @param directory true if to create a directory, false if file
+     */
+    private void createLocalFile(String path, boolean directory) {
+        boolean absolute;
+        String currentDirectory = filePanel.getCurrentWorkingDirectory();
+        try {
+            ResolvedPath resolvedPath = UI.resolveLocalPath(path, currentDirectory);
+            path = resolvedPath.getResolvedPath();
+            absolute = resolvedPath.isPathAlreadyAbsolute();
+        } catch (IOException ex) {
+            UI.doException(ex, UI.ExceptionType.EXCEPTION, FTPSystem.isDebugEnabled()); //this could keep happening, so show exception dialog
+            return;
+        }
+
+        String parentPath = UI.getParentPath(path);
+        boolean existsAsDir = new File(parentPath).isDirectory();
+
+        if (!existsAsDir) {
+            UI.doError("Directory does not exist", "Cannot create directory as path: " + parentPath + " does not exist");
+        } else {
+            if (directory) {
+                createLocalDirectory(path);
+            } else {
+                createLocalNormalFile(path);
+            }
+
+            boolean parentPathMatchesPanelsPath = currentDirectory.equals(parentPath);
+            if (!absolute && parentPathMatchesPanelsPath) {
+                filePanel.refresh(); // only need to refresh if the path was relative (as the directory would be created in the current folder) or if absolute and the prent path doesnt match current path. The path identified by the absolute will be refreshed when its navigated to
+            } else if (parentPathMatchesPanelsPath) {
+                filePanel.refresh();
+            }
+        }
+    }
+
+    /**
+     * The handler for creating a new directory
+     */
+    @Override
+    void createNewDirectory() {
+        String path = UI.doPathDialog(UI.PathAction.CREATE, true);
+
+        if (path != null)
+            createLocalFile(path, true);
+    }
+
+    /**
+     * The handler to create a new empty file
+     */
+    @Override
+    void createNewFile() {
+        String path = UI.doPathDialog(UI.PathAction.CREATE, false);
+
+        if (path != null)
+            createLocalFile(path, false);
+    }
+
+    /**
+     * Takes the given path and attempts to go the the location in the local file system identified by it.
+     * @param path the path to go to
+     */
+    private void goToLocalPath(String path) throws FileSystemException {
+        try {
+            ResolvedPath resolvedPath = UI.resolveLocalPath(path, filePanel.getCurrentWorkingDirectory());
+            path = resolvedPath.getResolvedPath();
+
+            LocalFile file = new LocalFile(new File(path).getAbsoluteFile().getPath());
```

Comment:
Changed this to just new LocalFile(path) as UI.resolveLocalPath returns an absolute (and canonical) path anyway